server/src/main/java/com/cloud/hypervisor/KVMGuru.java

```diff
@@ -132,7 +132,7 @@ public VirtualMachineTO implement(VirtualMachineProfile vm) {
             guestOsMapping = _guestOsHypervisorDao.findByOsIdAndHypervisor(guestOS.getId(), getHypervisorType().toString(), host.getHypervisorVersion());
         }
         if (guestOsMapping == null || host == null) {
-            to.setPlatformEmulator("Other");
+            to.setPlatformEmulator(guestOS.getDisplayName());
```

Comment:
@ustcweizhou how about we check and fallback to `"Other"` in case `guestOS.getDisplayName()` is null or empty (unlikely but just to make code more defensive)?